src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1518,7 +1518,10 @@ private File createWindowsSshAskpass(SSHUserPrivateKey sshUser) throws IOExcepti
         PrintWriter w = null;
         try {
             w = new PrintWriter(ssh);
-            w.println("echo \"" + Secret.toString(sshUser.getPassphrase()) + "\"");
+            // avoid echoing command as part of the password
+            w.println("@echo off");
+            // no need for quotes on windows echo -- they will get echoed too
+            w.println("echo " + Secret.toString(sshUser.getPassphrase()));
```

Comment:
This code only seems to be invoked when a passphrase has been assigned to the private key.  I was hoping it was the RSA private key (rather than the passphrase for the key) because that would mean it was encoding user input rather than echoing the user input without any encoding.  My hopes were wrong.

Since the resulting string from Secret.toString(sshUser.getPassPhrase()) is not encoded, it may contain one or more of the [Windows command processor special characters](http://ss64.com/nt/syntax-cmd.html) '&', '<', '>'. '(',  ')', '@', '^', '|'.

If the user provided passphrase includes one or more of those special characters, then the surrounding double quotes are needed (at least per the "Quote Characters in a command" section of the [ss64 command syntax site](http://ss64.com/nt/syntax-cmd.html).

If the user provided passphrase does not include one or more of those special characters, then the surrounding double quotes are not needed (and probably must not be included).

The original bug report [JENKINS-20356](https://issues.jenkins-ci.org/browse/JENKINS-20356) made no mention of an ssh passphrase being required to see the bug, yet this change seems to only be executed if the user provided an ssh passphrase protected private key.  

Does your use case require a passphrase protected private key?